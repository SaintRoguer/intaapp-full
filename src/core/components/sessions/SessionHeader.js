import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import moment from 'moment';
import 'moment/locale/es';
import {ReadMoreText} from '../../../shared/components/ReadMoreText';

export function SessionHeader({item: {date, user, description, visibility}}) {
  return (
    <View style={styles.inputContainer}>
      <>
        <Text>
          <Text style={styles.boldText}>Fecha:</Text>{' '}
          {date instanceof Date
            ? moment(date).format('L')
            : moment(date.toDate()).format('L')}
        </Text>
        <Text>
          <Text style={styles.boldText}>Creada por:</Text> {user}
        </Text>
        <Text>
          <Text style={styles.boldText}>Descripci√≥n: </Text>
        </Text>
        <ReadMoreText text={description} />
      </>
    </View>
  );
}
const styles = StyleSheet.create({
  inputContainer: {
    margin: 20,
    width: '100%',
  },
  boldText: {
    fontWeight: 'bold',
  },
});
