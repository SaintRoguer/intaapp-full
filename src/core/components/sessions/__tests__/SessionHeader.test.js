/**
 * @format
 */
import 'react-native';
import React from 'react';
import {SessionHeader} from '../SessionHeader.js';

import renderer from 'react-test-renderer';

describe('SessionDetail component', () => {
  test('Renders correctly', () => {
    const component = renderer.create(
      <SessionHeader
        item={{date: {toDate: () => new Date()}, user: 'Juan', description: ''}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
