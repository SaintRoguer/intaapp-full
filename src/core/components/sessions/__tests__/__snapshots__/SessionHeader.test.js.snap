// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionDetail component Renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <Text>
    Fecha: 
    6 de septiembre de 2020
  </Text>
  <Text>
    Creador/a: 
    Juan
  </Text>
  <Text>
    Description: 
    
  </Text>
</View>
`;
